warehouse/redshift/redshift.go

```diff
@@ -21,35 +21,43 @@ import (
 	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
 )
 
-var (
+type Config struct {
 	setVarCharMax                 bool
 	dedupWindow                   bool
 	dedupWindowInHours            time.Duration
-	pkgLogger                     logger.Logger
 	skipComputingUserLatestTraits bool
 	enableDeleteByJobs            bool
-)
+}
+
+var pkgLogger logger.Logger
 
 func Init() {
-	loadConfig()
 	pkgLogger = logger.NewLogger().Child("warehouse").Child("redshift")
 }
 
-func loadConfig() {
-	setVarCharMax = config.GetBool("Warehouse.redshift.setVarCharMax", false)
-
-	config.RegisterBoolConfigVariable(false, &dedupWindow, true, "Warehouse.redshift.dedupWindow")
-	config.RegisterDurationConfigVariable(720, &dedupWindowInHours, true, time.Hour, "Warehouse.redshift.dedupWindowInHours")
-	config.RegisterBoolConfigVariable(false, &skipComputingUserLatestTraits, true, "Warehouse.redshift.skipComputingUserLatestTraits")
-	config.RegisterBoolConfigVariable(false, &enableDeleteByJobs, true, "Warehouse.redshift.enableDeleteByJobs")
+// NewHandle return a new redshift handle
+// TODO: We should probably inject logger from warehouse manager
+func NewHandle() Handle {
+	return Handle{
+		Config: Config{
+			setVarCharMax:                 config.GetBool("Warehouse.redshift.setVarCharMax", false),
+			dedupWindow:                   config.GetBool("Warehouse.redshift.dedupWindow", false),
+			skipComputingUserLatestTraits: config.GetBool("Warehouse.redshift.skipComputingUserLatestTraits", false),
+			enableDeleteByJobs:            config.GetBool("Warehouse.redshift.enableDeleteByJobs", false),
+			dedupWindowInHours:            config.GetDuration("Warehouse.redshift.dedupWindow", 720, time.Hour),
+		},
+		Logger: pkgLogger,
+	}
```

Comment:
Our constructor is coupled with the config package.

I would suggest something like:
```suggestion
func WithConfig(h *Handle, config.Config) {
			h.SetVarCharMax =              config.GetBool("Warehouse.redshift.setVarCharMax", false)
			h.DedupWindow = config.GetBool("Warehouse.redshift.dedupWindow", false)                  
			h.SkipComputingUserLatestTraits = config.GetBool("Warehouse.redshift.skipComputingUserLatestTraits", false),
			h.EnableDeleteByJobs =          config.GetBool("Warehouse.redshift.enableDeleteByJobs", false),
			h.DedupWindowInHours =            config.GetDuration("Warehouse.redshift.dedupWindow", 720, time.Hour),
}
```
Usage:
```go
h := redshift.Handle{
   Logger: pkgLogger,
}
redshift.WithConfig(h, config.DefaultConfig)
```

Reply:
What do you think of adding WithConfig in the receiver itself. @lvrach